apps/devhub/tests/test_views.py

```diff
@@ -1207,6 +1212,23 @@ def test_delete_and_recreate_credentials(self):
         assert new_key.key != old_key.key
         assert new_key.secret != old_key.secret
 
+    def test_retreive_credentials(self):
+        client = Client(enforce_csrf_checks=True)
```

Comment:
oh, duh, I figured it out. CSRF is only used for POST, not GET. I know that it would make sense to use a GET to access the secret but you should make it require a POST so that 1) you get CSRF protection and 2) to prevent any caching that sometimes happens with GETs. After making it a post you'll be able to write a test using `enforce_csrf_checks=True` that will assert the response is a 403.